code/datums/martial/buster_style.dm

```diff
@@ -0,0 +1,493 @@
+			walk_towards(F, B, 0, 1.5)
+			animate(F, alpha = 0, color = "#d40a0a", time = 0.5 SECONDS) // Cool after-image
+			for(var/mob/living/L in T.contents) // Take all mobs we encounter with us
+				if(L != B) 
+					B.apply_status_effect(STATUS_EFFECT_DOUBLEDOWN)	
+					mopped |= L // Add them to the list of things we are mopping
+					L.add_fingerprint(B, FALSE)
+					var/turf/Q = get_step(get_turf(B), B.dir) // get the turf behind the thing we're attacking
+					to_chat(L, span_userdanger("[B] grinds you against the ground!"))
+					footsies(L)
+					if(istype(T, /turf/open/space)) // If we're about to hit space, throw the first mob into space
+						var/atom/throw_target = get_edge_target_turf(L, B.dir)
+						wakeup(L)
+						L.throw_at(throw_target, 2, 4, B, 3) // throwing them outside
+					if(Q.density) // If we're about to hit a wall
+						wakeup(L)
+						grab(B, L, crashdam) 
+						B.visible_message(span_warning("[B] rams [L] into [Q]!"))
+						to_chat(L, span_userdanger("[B] rams you into [Q]!"))
+						L.Knockdown(1 SECONDS)
+						L.Immobilize(1.5 SECONDS)
+						return // Then stop here
+					for(var/obj/D in Q.contents) // If we're about to hit a dense object like a table or window
+						wakeup(L)
+						if(D.density == TRUE)
+							grab(B, L, crashdam) 
+							B.visible_message(span_warning("[B] rams [L] into [D]!"))
+							to_chat(L, span_userdanger("[B] rams you into [D]!"))
+							D.take_damage(200) // Damage dense object
+							L.Knockdown(1 SECONDS)
+							L.Immobilize(1 SECONDS)
+							if(D.density == TRUE) // If it wasn't destroyed, stop here
+								return
+					B.forceMove(get_turf(L)) // Move buster arm user (forward) on top of the mopped mob
+					to_chat(L, span_userdanger("[B] catches you with [B.p_their()] hand and drags you down!"))
+					B.visible_message(span_warning("[B] hits [L] and drags them through the dirt!"))
+					L.forceMove(Q) // Move mopped mob forward
+					grab(B, L, dragdam) 
+					playsound(L,'sound/effects/meteorimpact.ogg', 60, 1)
+			T = get_step(B, B.dir) // Move our goalpost forward one
+	for(var/mob/living/C in mopped) // Return everyone to standing if they should be
+		wakeup(C)
+
+/*---------------------------------------------------------------
+	end of mop section
+---------------------------------------------------------------*/
+
+/*---------------------------------------------------------------
+	start of slam section
+---------------------------------------------------------------*/
+
+/datum/martial_art/buster_style/proc/slam(mob/living/user, mob/living/target)
+	var/supdam = 20
+	var/crashdam = 10
+	var/walldam = 30
+	var/mob/living/B = user
+	var/mob/living/L = target
+	var/turf/Z = get_turf(B)
+	if(next_slam > world.time)
+		to_chat(user, span_warning("You can't do that yet!"))
+		return
+	next_slam = world.time + COOLDOWN_SLAM
+	B.apply_status_effect(STATUS_EFFECT_DOUBLEDOWN)	
+	var/turf/Q = get_step(get_turf(B), turn(B.dir,180)) // Get the turf behind us
```

Comment:
```suggestion
	var/turf/flank_turf = get_step(get_turf(B), turn(B.dir,180)) // Get the turf behind us
```
this applies to everything like come on who is ABCDQ this is not the alphabet this is a game